pkg/build/gobuild.go

```diff
@@ -285,7 +288,12 @@ func build(ctx context.Context, ip string, dir string, platform v1.Platform, con
 
 	args = append(args, "-o", file)
 	args = append(args, ip)
-	cmd := exec.CommandContext(ctx, "go", args...)
+
+	gobin := defaultGoBin
```

Comment:
This is repeated enough maybe should have it be a buildOption, WithGoBinary("gotip")?

Reply:
Are you referring to this type def `BuildOptions` or creating a new one and make it a functional options parameter to those two functions so I don't have to change the whole call trees/callers? I would still have to check for set build option in those cases then unless I propagate a default (requires changing the call tree).

Without `pkd/commands.go` we have three copies of this code snippet currently.

Reply:
I simplified to a helper function, ok?